napari/_qt/qt_main_window.py

```diff
@@ -417,6 +420,21 @@ def __init__(self, viewer, *, show: bool = True):
         self.window_menu.addSeparator()
 
         settings.appearance.events.theme.connect(self._update_theme)
+        settings.application.events.grid_stride.connect(
```

Comment:
by connecting these events here on the `qt_main_window`, we're involving Qt in a "pure model connection"...  meaning these two will go out of sync if there is no window (i.e. a headless environment).  For example, the following code will give you a different result before/after running `pip uninstall qtpy`

```py
In [1]: import napari

In [2]: from napari.utils.settings import get_settings

In [3]: viewer = napari.Viewer()

In [4]: settings = get_settings()

In [5]: settings.application.grid_stride
Out[5]: 1

In [6]: viewer.grid.stride
Out[6]: 1

In [7]: settings.application.grid_stride = 4

In [8]: settings.application.grid_stride
Out[8]: 4

In [9]: viewer.grid.stride  # out of sync
Out[9]: 1
```

It's true that this would probably not be noticed in a headless environment (since there would be no visual representation of the grid)... but we should generally try to avoid requiring qt-in-the-loop for things that don't require Qt.

Reply:
I see. I moved the updates to viewer_model.py. Let me know if this works.  thanks!